corehq/apps/api/resources/v0_5.py

```diff
@@ -935,3 +935,48 @@ def prepend_urls(self):
             url(r"^(?P<resource_name>%s)/(?P<case_type>[\w\d_.-]+)/$" % self._meta.resource_name,
                 self.wrap_view('dispatch_list'))
         ]
+
+ODATA_XFORM_INSTANCE_RESOURCE_NAME = 'Forms'
+
+
+class ODataXFormInstanceResource(v0_4.XFormInstanceResource):
+
+    xmlns = None
+
+    def dispatch(self, request_type, request, **kwargs):
+        if not toggles.ODATA.enabled_for_request(request):
+            raise ImmediateHttpResponse(response=HttpResponseNotFound('Feature flag not enabled.'))
+        self.app_id = kwargs['app_id']
+        self.xmlns = kwargs['xmlns']
+        return super(ODataXFormInstanceResource, self).dispatch(request_type, request, **kwargs)
+
+    def determine_format(self, request):
+        # json only
+        return 'application/json'
+
+    def create_response(self, request, data, response_class=HttpResponse, **response_kwargs):
+        data['domain'] = request.domain
+        data['app_id'] = self.app_id
+        data['xmlns'] = self.xmlns
+        data['api_path'] = request.path
+        response = super(ODataXFormInstanceResource, self).create_response(
+            request, data, response_class, **response_kwargs)
+        return add_odata_headers(response)
+
+    def obj_get_list(self, bundle, domain, **kwargs):
+        elastic_query_set = super(ODataXFormInstanceResource, self).obj_get_list(bundle, domain, **kwargs)
+        full_xmlns = 'http://openrosa.org/formdesigner/' + kwargs['xmlns']
+        elastic_query_set.payload['filter']['and'].append({'term': {'xmlns.exact': full_xmlns}})
```

Comment:
ouch. I guess this doesn't use `ESQuery`?

Reply:
Nope :(